docs/develop-an-agent.md

```diff
@@ -0,0 +1,423 @@
+fi
+
+```
+
+> **Note**
+>
+> Portions of the script above come from the [iothub-get-twin.sh](https://github.com/Azure/iot-identity-service/blob/main/ci/iothub-get-twin.sh) script used in the tests for the identity service.
+
+
+#### **Step 5** Connect to IoT Hub using the SDK
+
+Connect to IoT Hub using the C SDK's [IoTHubDeviceClient_LL_CreateFromConnectionString](https://docs.microsoft.com/en-us/azure/iot-hub/iot-c-sdk-ref/iothub-device-client-ll-h/iothubdeviceclient-ll-createfromconnectionstring) API.
+
+#### **Step 6** After disconnect,
+
+Repeat all steps starting from _Step 1_ above.
+
+## Using the C SDK to connect to IoT Hub for SaS authentication
+
+The steps below are how to use the C SDK in your agent to establish IoT Hub connectivity using a module identity and using the SharedAccessSignature connection string method.
+
+The general SDK flow is as follows:
+
+-   Get identity
+    -   Set REST API URL
+        -   /identities/identity
+    -   Set corresponding JSON payload
+        -   No payload for identity
+    -   Send request to identity endpoint `/run/aziot/identityd.sock`
+        -   If the connection is refused, retry after few seconds
+    -   On a successful response, parse response (example response in the flow diagram)
+        -   Hub hostname
+        -   Gateway host name
+        -   Device Id
+        -   Module Id
+        -   Key handle
+-   Build a base64-encoded resource uri
+    -   Using process defined [above](#Step-2-Build-a-bas64-encoded-resource-uri)
+-   Get signed token
+    -   Set REST API URL
+        -   `/sign`
+    -   Set corresponding JSON payload
+        -   JSON payload for /sign:
+            -   Key handle from Get identity call above
+            -   Algorithm: "HMAC-SHA256"
+            -   Parameters:
+                -   Message: "&lt;base64-encoded resource uri&gt;"
+    -   Send request for key service endpoint `/run/aziot/keyd.sock`
+        -   If connection refused, go back to Get identity step above (since IoT Hub device information could have changed)
+    -   On successful response, parse response
+        -   SaS token
+-   Create a `sas_token` following these patterns:
+    -   Device e.g. `"SharedAccessSignature sr=myhubname.azure-devices.net/devices/mydeviceid&sig=hIuFfERqcDBnu84EwVlF01JfiaRvH6A20dMWQW6T4fg%3D&se=1609459200"`
+    -   Module e.g. `"SharedAccessSignature sr=myhubname.azure-devices.net/devices/mydeviceid/modules/mymoduleid&sig=hIuFfERqcDBnu84EwVlF01JfiaRvH6A20dMWQW6T4fg%3D&se=1609459200"`
+    -   Note that `sig` needs to be percent-encoded, but `sr` does not
+    -   Also note that the `SharedAccessSignature` is part of both the parameter name AND the parameter value in the connection string below. So,
+        -   `"SharedAccessSignature=SharedAccessSignature sr=..."`
+-   Build IoT Hub connection string with responses from two API calls above
+    -   `"HostName=<host_name>;DeviceId=<device_id>;SharedAccessSignature=<sas_token>"`
```

Comment:
Do you not need the HTML codes (gt; and lt;) here? If not, why use them on line 386?